lib/nerves_runtime.ex

```diff
@@ -68,6 +68,21 @@ defmodule Nerves.Runtime do
     end
   end
 
+  @doc """
+  Uses `boardid` to return a unique serial number.
+  """
+  @spec serial_number() :: String.t() | no_return()
+  def serial_number() do
+    case System.find_executable("boardid") do
```

Comment:
I'm for hardcoding the path to `"/usr/bin/boardid"` since it would be really confusing if this ever picked up an alternative. If someone needs it to be in a different location, I think that we could address that in a follow-up PR by adding an application config key. I think that will be rare.